### PR TITLE
Adding PHPUnit4To6Shim

### LINKED ISSUE #29: Add a polyfill for phpunit 4 to 6
https://github.com/GoogleCloudPlatform/php-docs-samples/pull/530/commits/f43e5e304d38bc58f83d64ea02771dcd32548360

### PR DESCRIPTION
**Pull request by tmatsuo**
Fixes #29 #30

### CODE REVIEW

**Comment by tmatsuo** [comment]
It fails as I expect

```
PHP Fatal error:  Class 'PHPUnit_Framework_TestCase' not found in /home/travis/build/GoogleCloudPlatform/php-tools/test/Utils/ContainerExecTest.php on line 31
```

File: src/TestUtils/EventuallyConsistentTestTrait.php [hunk 1]
```diff
@@ -17,7 +17,12 @@
 
 namespace Google\Cloud\TestUtils;
 
-use PHPUnit_Framework_ExpectationFailedException;
+if (!class_exists('\PHPUnit\Framework\ExpectationFailedException', true)) {
```
Comment (chingor13) [inline src/TestUtils/EventuallyConsistentTestTrait.php]: Does this belong in `src/TestUtils/PHPUnit4To6Sim.php` too?

Comment (tmatsuo) [reply src/TestUtils/EventuallyConsistentTestTrait.php]: This patch should work without installing the sim, so this is here.
This file should be usable for phpunit4 as well as phpunit6, without installing the sim.

Comment (chingor13) [reply src/TestUtils/EventuallyConsistentTestTrait.php]: ok

**Comment by tmatsuo** [comment]
@chingor13 Thanks, renamed to shim. PTAL

File: src/TestUtils/EventuallyConsistentTestTrait.php (hunk 1 above)
Comment (bshaffer) [inline src/TestUtils/EventuallyConsistentTestTrait.php]: Shouldn't this be done in the `bootstrap.php` file? Putting it at the top of a class file is pretty dirty.

File: test/bootstrap.php [hunk 2]
```diff
@@ -17,3 +17,7 @@
 
 require_once __DIR__ . '/../vendor/autoload.php';
 require_once __DIR__ . '/TestUtils/sleeper.php';
+
+use Google\Cloud\TestUtils\PHPUnit4To6Sim;
```
Comment (bshaffer) [inline test/bootstrap.php]: Do you mean `PHPUnit4to6Shim`? Why don't we just call it `PHPUnitShim`? Or have two different ones, `PHPUnit4Shim` and `PHPUnit6Shim`?

File: src/TestUtils/EventuallyConsistentTestTrait.php (hunk 1 above)
Comment (tmatsuo) [reply src/TestUtils/EventuallyConsistentTestTrait.php]: I wanted this file to be usable for both phpunit4 users and phpunit6 users, without adding anything in the bootstrap file.

Comment (bshaffer) [reply src/TestUtils/EventuallyConsistentTestTrait.php]: A different way to do this then would be to catch all exceptions and check the classname, or alternately add a method inside the trait such as `aliasExpectationFailedException` and call it before the try/catch.

**Pull request merged**